go_expvar/datadog_checks/go_expvar/go_expvar.py

```diff
@@ -226,18 +227,31 @@ def deep_get(self, content, keys, traversed_path=None):
         if keys == []:
             return [(traversed_path, content)]
 
+        results = []
+
         key = keys[0]
-        regex = "".join(["^", key, "$"])
-        try:
-            key_rex = re.compile(regex)
-        except Exception:
-            self.warning("Cannot compile regex: %s" % regex)
-            return []
+        if key.isalnum():
+            # key is not a regex, simply match for equality
+            for new_key, new_content in self.items(content):
+                if key == new_key:
+                    results.extend(self.deep_get(new_content, keys[1:], traversed_path + [str(new_key)]))
```

Comment:
Instead of duplicating the loop, I propose something like that:
```
if key.isalnum():
   match = key.__eq__
else:
   match = re.compile(...).match
```

Reply:
Neat! I didn't find the way to not copy the loop :)